stackrox/stackrox
# Check public config on analytics events

## Description

Whenever analytics methods are called, this option checks the public config. If telemetry is disabled in the public config, events will not be sent to Segment.

Caveat: If telemetry collection is enabled when a user visits the page, Segment is initialized. If the user decides to opt out of analytics, events will no longer be sent to Segment, but Segment will still be set up on the page unless we manually remove the script tags and delete `window.analytics`, which is not recommended.

## Checklist

If any of these don't apply, please comment below.

## Testing Performed

With the flag enabled, you can verify segment is initialized in various ways:
* check segments debugger and look for events
* see if window.analytics exists
* look for segment/amplitude script tags

Disabling telemetry in the system config should result in events not being sent to segment, however `window.analytics` will still exist and the script tags can still be viewed in the html
